### PR TITLE
Tech debt

### PR DESCRIPTION
**Pull request by rustworthy**
- Use ::connect() vs ::connect_to(&str) as per [thread](https://github.com/jonhoo/faktory-rs/pull/64#discussion_r1720922173)
 
- Job's `reserve_for` is now using Duration;

- We are adding public  `Wokrer::is_terminated` so that they can check and _not_ run a terminated worker;

- Refactoring: using `tokio::time::interval` in the heartbeat thread instead of `tokio::time::sleep`

- - -
This change is [Reviewable](https://reviewable.io/reviews/jonhoo/faktory-rs/79)

### CODE REVIEW

**Comment by rustworthy** [comment]
@jonhoo Sorry it took me a while this time. I was moving apartment and preparing some legal docs around my expat life :sweat_smile: 

So I have addressed  - I _think_ so - all the threads, but [this one](https://github.com/jonhoo/faktory-rs/pull/64#discussion_r1739996721). I was guided by the compiler and the current implementation made it happy, and you unhappy :smile:  I agree it looks and feels weird, but I think I will need your help with this.

Please let me know if you while reviewing you catch other things that we better refactor now if this will be breaking.

File: src/proto/single/utils.rs
Comment (jonhoo) [inline src/proto/single/utils.rs:33]:
```suggestion
pub(crate) fn ser_optional_duration_in_seconds<S>(
```
the "in seconds" part is the main bit that requires this helper function. ditto for `deser`.

**Review by jonhoo — APPROVED** [review]
These changes all look great, thanks! A small nit is all.

No worries with the delay — I've had a similar delay now, and will likely be gone for the next month or so between work and holidays :sweat_smile:

**Comment by jonhoo** [comment]
Feel free to merge this yourself once corrected :+1: 

I've also now given you publish rights to `faktory` on crates.io — you're now a full-fledged maintainer :)

**Comment by jonhoo** [comment]
On the `Reconnect` thing, it's probably not ultimately all that important. My main wonder is if we can replace
```rust
impl Reconnect for BufStream<TlsStream<tokio::net::TcpStream>>
impl Reconnect for BufStream<TlsStream<proto::BoxedConnection>>
```
with
```rust
impl<S> Reconnect for TlsStream<S> where S: Reconnect
```

I'm also a little suspicious of
```rust
impl Reconnect for BufStream<TlsStream<proto::BoxedConnection>>
```
in the first place since that will end up returning a `BoxedConnection` that in turn _contains_ a `BoxedConnection`, meaning an extra (unnecessary) indirection.

**Pull request merged**
